### PR TITLE
feat(sampling): Add rate suggestion UI [TET-164]

### PR DESCRIPTION
**Pull request by matejminar**
https://user-images.githubusercontent.com/9060071/177207316-abaf4a0a-de64-41a2-912b-6ab72b84b1ff.mp4

There is a good chance that the logic for the rate suggestion / chart prediction will change.
I did not handle some edge-cases / clamping on purpose so that we can see where this logic falls short more easily.

### CODE REVIEW

**Review by andriisoldatenko — APPROVED** [review]

File: static/app/views/settings/project/server-side-sampling/utils/index.tsx [hunk 1]
```diff
@@ -92,3 +92,5 @@ export function getInnerNameLabel(name: SamplingInnerName | string) {
       return `${stripCustomTagPrefix(name)} - ${t('Custom')}`;
   }
 }
+
+export const field = 'sum(quantity)';
```
Comment (priscilawebdev) [inline static/app/views/settings/project/server-side-sampling/utils/index.tsx:96]: should we make this const more descriptive?

File: static/app/views/settings/project/server-side-sampling/utils/projectStatsToSeries.tsx [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        case Outcome.INVALID:
+        case Outcome.RATE_LIMITED:
+          seriesData.droppedServer[index] = {
+            name: moment(interval).valueOf(),
+            value:
+              (seriesData.droppedServer[index]?.value ?? 0) + group.series[field][index],
+          };
+          break;
+        default:
+        //
```
Comment (priscilawebdev) [inline static/app/views/settings/project/server-side-sampling/utils/projectStatsToSeries.tsx:59]: should we throw an exception or something?

File: tests/js/spec/views/settings/project/server-side-sampling/utils/projectStatsToSampleRates.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import {projectStatsToSampleRates} from 'sentry/views/settings/project/server-side-sampling/utils/projectStatsToSampleRates';
+
+describe('projectStatsToSampleRates', () => {
+  it('basic', () => {
```
Comment (priscilawebdev) [inline tests/js/spec/views/settings/project/server-side-sampling/utils/projectStatsToSampleRates.spec.tsx:4]:
```suggestion
  it('basic', function()  {
```

File: tests/js/spec/views/settings/project/server-side-sampling/utils/projectStatsToSampleRates.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import {projectStatsToSampleRates} from 'sentry/views/settings/project/server-side-sampling/utils/projectStatsToSampleRates';
+
+describe('projectStatsToSampleRates', () => {
```
Comment (priscilawebdev) [inline tests/js/spec/views/settings/project/server-side-sampling/utils/projectStatsToSampleRates.spec.tsx:3]:
```suggestion
describe('projectStatsToSampleRates', function() {
```

File: tests/js/spec/views/settings/project/server-side-sampling/utils/projectStatsToSeries.spec.tsx [hunk 5]
```diff
@@ -0,0 +1,179 @@
+import {projectStatsToSeries} from 'sentry/views/settings/project/server-side-sampling/utils/projectStatsToSeries';
+
+describe('projectStatsToSeries', () => {
+  it('basic', () => {
```
Comment (priscilawebdev) [inline tests/js/spec/views/settings/project/server-side-sampling/utils/projectStatsToSeries.spec.tsx:4]: basic?

Comment (priscilawebdev) [inline tests/js/spec/views/settings/project/server-side-sampling/utils/projectStatsToSeries.spec.tsx:4]:
```suggestion
describe('projectStatsToSeries', function() {
  it('basic', function() {
```

**Comment by priscilawebdev** [comment]
should we make the modal fit on smaller devices?

https://user-images.githubusercontent.com/29228205/177279982-12bb16d9-602b-4751-baf6-2e84cc448352.mov

File: tests/js/spec/views/settings/project/server-side-sampling/utils/projectStatsToSampleRates.spec.tsx (hunk 3 above)
Comment (matejminar) [reply tests/js/spec/views/settings/project/server-side-sampling/utils/projectStatsToSampleRates.spec.tsx]: I updated it but I don't think stylistic preferences should be the topic of code review - if we care about having it a specific way, we should create a lint rule.

Comment (priscilawebdev) [reply tests/js/spec/views/settings/project/server-side-sampling/utils/projectStatsToSampleRates.spec.tsx]: it was just a suggestion 😉

**Pull request merged**